addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js

```diff
@@ -177,22 +173,53 @@ class BudgetsService extends Service {
   }
 
   _formBudgetObject(newBudgetConfig) {
-    const budget = this.budgetBaseTemplates.BudgetTemplate;
-    budget.BudgetName = this.budgetName;
-    budget.BudgetLimit.Amount = newBudgetConfig.budgetLimit;
-    budget.TimePeriod.Start = newBudgetConfig.startDate;
-    budget.TimePeriod.End = newBudgetConfig.endDate;
+    const budget = {
+      BudgetName: this.budgetName,
+      BudgetType: 'COST',
+      TimeUnit: 'ANNUALLY',
+      BudgetLimit: {
+        Amount: newBudgetConfig.budgetLimit,
+        Unit: 'USD',
+      },
+      CostTypes: {
+        IncludeCredit: true,
+        IncludeDiscount: true,
+        IncludeOtherSubscription: true,
+        IncludeRecurring: true,
+        IncludeRefund: true,
+        IncludeSubscription: true,
+        IncludeSupport: true,
+        IncludeTax: true,
+        IncludeUpfront: true,
+        UseAmortized: true,
+        UseBlended: false,
+      },
+      TimePeriod: {
+        End: newBudgetConfig.endDate,
+        Start: newBudgetConfig.startDate,
+      },
+    };
     return budget;
   }
 
   _formNotificationObjects(newBudgetConfig) {
     const notifications = [];
-    const notification = this.budgetBaseTemplates.NotificationTemplate;
     const thresholds = _.get(newBudgetConfig, 'thresholds', []);
     thresholds.forEach(threshold => {
-      const notificationCopy = _.cloneDeep(notification);
-      notificationCopy.Notification.Threshold = threshold;
-      notificationCopy.Subscribers[0].Address = newBudgetConfig.notificationEmail;
+      const notificationCopy = {
+        Notification: {
+          ComparisonOperator: 'GREATER_THAN',
+          NotificationType: 'ACTUAL',
+          Threshold: threshold,
+          ThresholdType: 'PERCENTAGE',
+        },
+        Subscribers: [
+          {
+            Address: newBudgetConfig.notificationEmail,
+            SubscriptionType: 'EMAIL',
+          },
+        ],
+      };
       notifications.push(notificationCopy);
     });
     return notifications;
```

Comment:
Minor. Since you are translating (mapping) thresholds to notifications it may be better to use `_.map` instead of `_.forEach`. This way, you don't need to keep `notifications` array and push to it. 

Something like,

```
_formNotificationObjects(newBudgetConfig) {
    const thresholds = _.get(newBudgetConfig, 'thresholds', []);
    return _.map(thresholds, threshold => {
      return {
        Notification: {
          ComparisonOperator: 'GREATER_THAN',
          NotificationType: 'ACTUAL',
          Threshold: threshold,
          ThresholdType: 'PERCENTAGE',
        },
        Subscribers: [
          {
            Address: newBudgetConfig.notificationEmail,
            SubscriptionType: 'EMAIL',
          },
        ],
      };
    });
  }
```

Reply:
Good call! Updating it now.